### PR TITLE
Add is0gr to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding the is0gr PCB to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/is0gr/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2023 Andrew Kannan
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/cannonkeys/is0gr/keymaps/via/keymap.c:7]:
```suggestion
  [0] = LAYOUT_ortho_1x1(
```

File: keyboards/cannonkeys/is0gr/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2023 Andrew Kannan
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/cannonkeys/is0gr/keymaps/default/keymap.c:7]:
```suggestion
  [0] = LAYOUT_ortho_1x1(
```

File: keyboards/cannonkeys/is0gr/info.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    "processor": "STM32F072",
+    "url": "https://cannonkeys.com",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0xCA04",
+        "vid": "0x0028"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/cannonkeys/is0gr/info.json:27]:
```suggestion
    },
    "community_layouts": ["ortho_1x1"],
    "layouts": {
        "LAYOUT_ortho_1x1": {
```

**Review by drashna — APPROVED** [review]

File: keyboards/cannonkeys/is0gr/info.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "maintainer": "awkannan",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/cannonkeys/is0gr/info.json:13]: Don't know if it's still applicable, but enabling nkro historically improved the output speed of send_string.

[the remark this answers is not in the document]
Comment (awkannan) [reply keyboards/cannonkeys/is0gr/info.json:13]: Gonna keep nkro.
About command and console - sure they are disabled by default, and I have no problem removing them, but is it really an issue to keep them in as well?

[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/cannonkeys/is0gr/info.json]: Shouldn't matter either way, it's just a bit cleaner.

**Review by tzarc — APPROVED** [review]

**Pull request merged**